### PR TITLE
ts2.4 prework

### PR DESCRIPTION
**Pull request by evmar**
This series of commits makes adjustments to the code that are required for TS2.4, but that still work in TS2.3.  So they are safe to land on master as-is, without changing our TS version.

You can review each commit individually, might be easier.

### CODE REVIEW

**Comment by evmar** [comment]
@rkirov for review, @tbosch just FYI

File: src/tsickle.ts [hunk 1]
```diff
@@ -188,6 +188,17 @@ class ClosureRewriter extends Rewriter {
   }
 
   /**
+   * Get the ts.Symbol at a location or throw.
+   * The TypeScript API can return undefined when fetching a symbol, but
+   * in many contexts we know it won't (e.g. our input is already type-checked).
+   */
+  mustGetSymbolAtLocation(node: ts.Node): ts.Symbol {
+    const sym = this.typeChecker.getSymbolAtLocation(node);
+    if (!sym) throw new Error('no symbol');
```
Comment (rkirov) [inline src/tsickle.ts]: can you throw in some `node` info in the Error.

File: src/tsickle.ts [hunk 2]
```diff
@@ -256,6 +258,7 @@ class ClosureRewriter extends Rewriter {
       }
       // Merge the parameters into a single list of merged names and list of types
       const sig = typeChecker.getSignatureFromDeclaration(fnDecl);
+      if (!sig) throw new Error('invalid signature');
```
Comment (rkirov) [inline src/tsickle.ts]: add 'fnDecl' name to the error.

File: src/tsickle.ts (hunk 1 above)
Comment (evmar) [inline src/tsickle.ts]: If we throw while traversing the AST, we show the file, line:col.

File: src/tsickle.ts (hunk 2 above)
Comment (evmar) [inline src/tsickle.ts]: Done

**Pull request merged**